From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jesse Buhagiar <jooster669@gmail.com>
Date: Sun, 2 Jan 2022 00:39:02 +1100
Subject: [PATCH] Build: Add SerenityOS to list of compatible systems

This is required by the build system to spit out a library with
the correct name/platform.
---
 engine/common/build.c | 2 ++
 public/build.h        | 4 ++++
 2 files changed, 6 insertions(+)

diff --git a/engine/common/build.c b/engine/common/build.c
index c4ddaee..42ba572 100644
--- a/engine/common/build.c
+++ b/engine/common/build.c
@@ -95,6 +95,8 @@ const char *Q_buildos( void )
 	osname = "DOS4GW";
 #elif XASH_HAIKU
 	osname = "haiku";
+#elif XASH_SERENITY
+	osname = "serenityos";
 #else
 #error "Place your operating system name here! If this is a mistake, try to fix conditions above and report a bug"
 #endif
diff --git a/public/build.h b/public/build.h
index 6e1f326..57a7735 100644
--- a/public/build.h
+++ b/public/build.h
@@ -74,6 +74,7 @@ For more information, please refer to <http://unlicense.org/>
 #undef XASH_RISCV_DOUBLEFP
 #undef XASH_RISCV_SINGLEFP
 #undef XASH_RISCV_SOFTFP
+#undef XASH_SERENITY
 #undef XASH_WIN32
 #undef XASH_WIN64
 #undef XASH_X86
@@ -125,6 +126,9 @@ For more information, please refer to <http://unlicense.org/>
 #elif defined __HAIKU__
 	#define XASH_HAIKU 1
 	#define XASH_POSIX 1
+#elif defined __serenity__
+	#define XASH_SERENITY 1
+	#define XASH_POSIX 1
 #else
 #error "Place your operating system name here! If this is a mistake, try to fix conditions above and report a bug"
 #endif
